### ISSUE TITLE
Telephony services silently crash. Possibly WiFi related.

### ISSUE THREAD

**Issue by nsensfel** [ubports/ubuntu-touch#1614]
- **Device:** F(x)tec Pro1 (halium Generic Device)
- **Channel:** Release Candidate
- **Build:** 2020-W49/2
- **Vendor Build Date:** Tue Aug 25 23:14:33 CST 2020

#### Solution 1:
1. Wait an undetermined amount of time (only tested with Wi-Fi on).
2. Attempt to make a call.

#### Solution 2:
1. Boot the device with Wi-Fi enabled.
2. Start Phone application.
3. Make a call.
4. End the call.
5. Disable Wi-Fi.
6. Observe Unity8 rebooting.
7. (Optional: re-activate Wi-Fi)
8. Attempt to make another call.

### Expected behavior
The call goes through.

#### Wi-Fi is on (solution 1, or solution 2 without step 7):
The phone application shows the expected service provider. The call is attempted, then "Call failed" appears.

#### Wi-Fi is off (solution 2 with step 7):
The phone application shows that there is "No network". Calls cannot be attempted.

### Logfiles and additional information
Application legacy dialer app: http://paste.ubuntu.com/p/yrQxPzX8ZQ/
Telephony service indicator: http://paste.ubuntu.com/p/k4YvMyNVbv/

Information from Alfred, in the UBPorts Telegram channel:

> We've been seeing similar problems on the Pixel 3a. It seems to be a common issue among Qualcomm devices. Best case someone with the BSP sources (maybe fxtec themselves) steps up and finds a workaround without modifying qcrild.
> The problem is related to qcrild going out of Binder memory, not being able to communicate with ofono anymore. Restarting ofono helps in most cases

**Comment by nsensfel** [comment]
Any SMS that was supposed to be received while this bug is active appears to be lost. Quite possibly permanently: I have received an SMS during a call, with someone who had sent me an SMS earlier that I did not received. Thus, telephony being restored did not recover the missing SMS.

Additional logs:
- **dmesg:** https://paste.ubuntu.com/p/3Tr993NNgY/
- **logcat:** https://paste.ubuntu.com/p/4z38cQzBVw/

These two new logs were obtained for the following scenario:

1. Boot phone with Wi-Fi already on.
2. Make call.
3. End call.
4. Turn off Wi-Fi.
5. There is no longer any service available.

**Comment by nsensfel** [comment]
I did [a full firmware restore](https://community.fxtec.com/topic/2466-guide-restore-to-stock-firmware-using-fastboot-method/) followed by a clean install of Ubuntu Touch. The issue can no longer be reproduced. I am going to assume that I messed up flashing/updating the firmware at some point and that this was the cause for the bug.
I'll be testing telephony regularly thorough the week, to see if the issue appears again, but I'll close this report in the meantime as it does not appear to be reproducible for now.
